packages/backend/src/managers/playground/McpServerManager.ts

```diff
@@ -15,52 +15,67 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
-import type { Disposable } from '@podman-desktop/api';
-import { promises } from 'node:fs';
 import path from 'node:path';
-import { type McpSettings, type McpServer, McpServerType } from '../../models/mcpTypes';
+import { type Disposable, EventEmitter } from '@podman-desktop/api';
+import { MSG_MCP_SERVERS_UPDATE } from '@shared/Messages';
+import { type McpSettings, McpServerType, type McpClient } from '@shared/models/McpSettings';
+import type { RpcExtension } from '@shared/messages/MessageProxy';
+import { JsonWatcher } from '../../utils/JsonWatcher';
+import { Publisher } from '../../utils/Publisher';
+import { toMcpClients } from '../../utils/mcpUtils';
 
 // TODO: Agree on the name of the file and its location
 const MCP_SETTINGS = 'mcp-settings.json';
 
-// TODO: Consider using JsonWatcher in case the file is going to be updated externally
-// Depending on the use case, we might want to watch for changes in the settings file.
-// 1. As a user, I want to be able to modify the mcp-settings.json file and have the changes reflected in the application without restarting it.
-// 2. As a user, I want to be able to add/remove servers from the mcp-settings.json while also having the application to provide a UI to manage the servers.
-// 3. As a user, I don't want to tinker with the mcp-settings.json file and would like to have the application to provide a UI to manage the servers.
-// Cases 1 and 2 are covered by JsonWatcher, but case 3 is not.
-// For 2 and 3, we need to implement a UI to manage the servers.
-export class McpServerManager implements Disposable {
+export class McpServerManager extends Publisher<McpSettings> implements Disposable {
   private readonly settingsFile: string;
+  private mcpSettings: McpSettings;
+  private readonly _onUpdate: EventEmitter<McpSettings>;
+  readonly #jsonWatcher: JsonWatcher<McpSettings>;
 
-  constructor(private appUserDirectory: string) {
+  constructor(
+    rpcExtension: RpcExtension,
+    private appUserDirectory: string,
+  ) {
+    super(rpcExtension, MSG_MCP_SERVERS_UPDATE, () => this.getMcpSettings());
     this.settingsFile = path.join(this.appUserDirectory, MCP_SETTINGS);
+    this.mcpSettings = {
+      servers: {},
+    };
+    this._onUpdate = new EventEmitter<McpSettings>();
+    this.#jsonWatcher = new JsonWatcher<McpSettings>(this.settingsFile, { ...this.mcpSettings });
+    this.#jsonWatcher.onContentUpdated(this.onMcpSettingsUpdated.bind(this));
+    this.#jsonWatcher.init();
```

Comment:
we usually uses a `init()` method on managers class

https://github.com/containers/podman-desktop-extension-ai-lab/blob/878854ad6ddcc8d8551404b11fd124f595b81e96/packages/backend/src/managers/catalogManager.ts#L63

might be nice to init the JsonWatcher inside this method.

> The idea is that in the future, we only init classes when we need its content